src/SimpleCircuitBreaker.php

```diff
@@ -31,30 +31,30 @@ public function call($service, callable $fallback)
     {
         $transaction = $this->initTransaction($service);
 
-        try {
-            if ($this->isOpened()) {
-                if ($this->canAccessService($transaction)) {
-                    $this->moveStateTo(States::HALF_OPEN_STATE, $service);
-                }
-
-                return \call_user_func($fallback);
+        if ($this->isOpened()) {
```

Comment:
initially, I've put this call in try/catch because... of people that could "chain" calls of CB.

call some/service
 --> call some/other/service
    ->> call some/another/service
     ->> OK then a static function

Reply:
was it a bad idea?

Reply:
oh you mean if the callback uses another circuit breaker? I don't think it's a problem then each will catch its own exception
I think the if statement could be outside of the try catch, but it's not that important
I reverted that change, I needed it when using a while loop, but since the recursive method works it's not necessary any more